src/api/participantsRouter.ts

```diff
@@ -1,21 +1,49 @@
-import express from 'express';
+import express, { NextFunction, Request, Response } from 'express';
 import { z } from 'zod';
 
-import { Participant, ParticipantSchema } from './entities/Participant';
+import { Participant, ParticipantCreationPartial, ParticipantSchema } from './entities/Participant';
 import { UserRole } from './entities/User';
 import { getKcAdminClient } from './keycloakAdminClient';
 import { createNewUser, sendInviteEmail } from './services/kcUsersService';
 import { createUserInPortal, isUserBelongsToParticipant } from './services/usersService';
 
 export const participantsRouter = express.Router();
+
+const idParser = z.object({
+  participantId: z.coerce.number(),
+});
+
+export interface ParticipantRequest extends Request {
+  participant?: Participant;
+}
+
+export const hasParticipantAccess = async (
+  req: ParticipantRequest,
+  res: Response,
+  next: NextFunction
+) => {
+  const { participantId } = idParser.parse(req.params);
+  const participant = await Participant.query().findById(participantId);
+  if (!participant) {
+    return res.status(404).send([{ message: 'The participant cannot be found.' }]);
+  }
+
+  if (!(await isUserBelongsToParticipant(req.auth?.payload?.email as string, participantId))) {
+    return res.status(401).send([{ message: 'You do not have permission to update participant.' }]);
```

Comment:
This should be a 403. 401 is if the user has no/invalid credentials, and they should log in and try again. 403 is for a user who's logged in, but doesn't have permission to do the operation.